### PR TITLE
Add `ACCOUNT_SET_PASSWORD_REQUESTED` webhook event

### PR DESCRIPTION
**Pull request by Air-t**
Adds `ACCOUNT_SET_PASSWORD_REQUESTED` event sent after `requestPasswordReset` or `customerCreate` mutation.

Resolves https://github.com/saleor/saleor/issues/13486

# Impact

* [ ] New migrations
* [ ] New/Updated API fields or mutations
* [ ] Deprecated API fields or mutations
* [ ] Removed API types, fields, or mutations
* [ ] Documentation needs to be updated

# Pull Request Checklist

* [ ] Privileged queries and mutations are either absent or guarded by proper permission checks
* [ ] Database queries are optimized and the number of queries is constant
* [ ] Database migrations are either absent or optimized for zero downtime
* [ ] The changes are covered by test cases

### CODE REVIEW

File: saleor/graphql/account/mutations/authentication/request_password_reset.py [hunk 1]
```diff
@@ -99,25 +106,41 @@ def clean_user(cls, email, redirect_url, info: ResolveInfo):
     def perform_mutation(cls, _root, info: ResolveInfo, /, **data):
         email = data["email"]
         redirect_url = data["redirect_url"]
-        channel_slug = data.get("channel")
         user = cls.clean_user(email, redirect_url, info)
+        user.last_password_reset_request = timezone.now()
+        user.save(update_fields=["last_password_reset_request"])
```
Comment (maarcingebala) [inline saleor/graphql/account/mutations/authentication/request_password_reset.py:111]: I'm not sure if we should save `last_password_reset_request` here; we run some validations in `post_save_action`, so wouldn't that lead to a situation where the new timestamp is saved, and later we get a validation error, which would effectively not send the notification?

File: saleor/graphql/webhook/enums.py [hunk 2]
```diff
@@ -41,6 +41,9 @@
     WebhookEventAsyncType.ACCOUNT_CHANGE_EMAIL_REQUESTED: (
         "Account email change is requested."
     ),
+    WebhookEventAsyncType.ACCOUNT_SET_PASSWORD_REQUESTED: (
+        "Account set password is requested."
```
Comment (maarcingebala) [inline saleor/graphql/webhook/enums.py:45]: It sounds unnatural, I would rephrase it to something like "Setting a new password for the account is requested."

File: saleor/graphql/webhook/subscription_types.py [hunk 3]
```diff
@@ -214,6 +214,14 @@ def resolve_new_email(root, _info: ResolveInfo):
         return data["new_email"]
 
 
+class AccountSetPasswordRequested(SubscriptionObjectType, AccountOperationBase):
+    class Meta:
+        root_type = None
+        enable_dry_run = False
+        interfaces = (Event,)
+        description = "Event sent when set password is requested." + ADDED_IN_315
```
Comment (maarcingebala) [inline saleor/graphql/webhook/subscription_types.py:222]: "Event sent when setting a new password is requested." maybe like this?

File: saleor/graphql/account/mutations/authentication/request_password_reset.py [hunk 4]
```diff
@@ -47,7 +50,11 @@ class Meta:
             WebhookEventInfo(
                 type=WebhookEventAsyncType.NOTIFY_USER,
                 description="A notification for password reset.",
-            )
+            ),
+            WebhookEventInfo(
+                type=WebhookEventAsyncType.ACCOUNT_SET_PASSWORD_REQUESTED,
+                description="An Account set password is requested.",
```
Comment (IKarbowiak) [inline saleor/graphql/account/mutations/authentication/request_password_reset.py:56]:
```suggestion
                description="A notification for account set password is requested.",
```

File: saleor/graphql/account/mutations/staff/customer_create.py [hunk 5]
```diff
@@ -33,4 +33,8 @@ class Meta:
                 type=WebhookEventAsyncType.NOTIFY_USER,
                 description="A notification for setting the password.",
             ),
+            WebhookEventInfo(
+                type=WebhookEventAsyncType.ACCOUNT_SET_PASSWORD_REQUESTED,
+                description="An Account set password is requested.",
```
Comment (IKarbowiak) [inline saleor/graphql/account/mutations/staff/customer_create.py:38]:
```suggestion
                description="A notification for account set password is requested.",
```

**Review by IKarbowiak — APPROVED** [review]

File: saleor/graphql/account/mutations/authentication/request_password_reset.py (hunk 1 above)
Comment (Air-t) [reply saleor/graphql/account/mutations/authentication/request_password_reset.py:111]: If there are validations which I missed, while changing the order of sending notifications with instance save, I will need to send notifications before save is done. However this will not be post_save_actions anymore (triggered before the actual save)

Comment (Air-t) [reply saleor/graphql/account/mutations/authentication/request_password_reset.py:112]: @maarcingebala I have added changes, however I'm not sure if sending a notification and triggering webhooks should still be a part of post_save_actions - where they are triggered before save. Should I change it, moving them to separate function or just to save itself?

Comment (maarcingebala) [reply saleor/graphql/account/mutations/authentication/request_password_reset.py:112]: Sending the event is the main point of this mutation, not a save. So I think only the naming "post_save" is confusing here. Maybe we don't need this function at all and everything should happen perform_mutation.

Comment (maarcingebala) [reply saleor/graphql/account/mutations/authentication/request_password_reset.py:112]: We're overriding `perform_mutation` anyway, and `post_save_action` is mostly useful when you want to override just that "post save" part of the flow.

Comment (Air-t) [reply saleor/graphql/account/mutations/authentication/request_password_reset.py:112]: My idea was to keep additional actions in one place, however maybe for this case it does not make sense to add extra method which naming is misleading just to keep all events calls in `post_save_action` as a pattern

Comment (Air-t) [reply saleor/graphql/account/mutations/authentication/request_password_reset.py]: anyways, I added all the logic to the `perform_mutation` method cause it seems to be more convenient

**Review by korycins — APPROVED** [review]

**Review by maarcingebala — APPROVED** [review]

**Comment by maarcingebala** [comment]
@Air-t Looks good, but the CI is failing here.

**Review by szczecha — APPROVED** [review]
